### PR TITLE
LPS-162830 Support automatic registration of ConfigurationBeanDeclaration services

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-162830](https://issues.liferay.com/browse/LPS-162830).

**tl;dr**: You do not have to create `ConfigurationBeanDeclaration` classes anymore. Just write your `*Configuration.java` class and it will automatically work with `ConfigurationProvider`.

---

This removes `ConfigurationBeanDeclaration` from `liferay-portal`. 

It has been replaced by a `BundleTracker` that will check each bundle as it is registered for Metatype information. If found, it will attempt to load each configuration class using the bundle's classloader. Any loaded classes that do not contain the proper annotations (`Meta.OCD` and `Meta.AD` with `required = false`) will be ignored. Otherwise, the relevant `ManagedService` and `ManagedServiceFactory` services will be registered.

On bundle deactivation, the services will be un-registered.

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
@dantewang @tinatian see https://github.com/liferay-usm/liferay-portal/pull/830/commits/5a43f92025c2185b1b9037513c6c3ea70dc1f418

**Comment by drewbrokke** [comment]
@stian-sigvartsen FYI https://github.com/liferay-usm/liferay-portal/pull/830/commits/ad4c8d51ed9ad0cf2f984a7dca848c49fc75796f

**Comment by drewbrokke** [comment]
ci:help

**Comment by drewbrokke** [comment]
ci:stop

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
@drewbrokke ad4c8d5 looks like a nice simplification. Will keep an eye out for issues in the CI testing.

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by melodywu** [comment]
ci:test:relevant

**Comment by melodywu** [comment]
No related failure found in ci:test:relevant. Because for [Staging](https://testray.liferay.com/home/-/testray/case_results/2035265546?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2035262597%26testrayRunId%3D2035262642%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc) and [File Install](https://testray.liferay.com/home/-/testray/case_results/2035265693?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2035262597%26testrayRunId%3D2035262642%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc) component Aborted prior to running test I run the ci:test:relevant  again

**Comment by melodywu** [comment]
Relevant failures are not related

**Comment by JoyceWang08211** [comment]
ci:test:mfa

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:mfa

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:reevaluate:1357401_3001

**Comment by drewbrokke** [comment]
ci:reevaluate:1347401_3159

**Comment by melodywu** [comment]
No related issue found in ci:test:mfa and ci:test:relevant

**Comment by Sharryxue** [comment]
ci:test:acceptance

**Comment by melodywu** [comment]
No related issue found in ci:test:acceptance

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
@patriciajeanperez if this looks good to you then I think we're good to forward.

**Comment by patriciajeanperez** [comment]
Pushed to BChan
https://github.com/brianchandotcom/liferay-portal/pull/126096
